yarn-project/noir-contracts/src/contracts/ecdsa_account_contract/src/main.nr

```diff
@@ -84,6 +60,56 @@ contract EcdsaAccount {
         );
     }
 
+    #[aztec(private)]
+    fn entrypoint(payload: pub EntrypointPayload) {
+        let actions = AccountActions::private(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.entrypoint(payload);
+    }
+
+    #[aztec(private)]
+    fn is_valid(message_hash: Field) -> Field {
+        let actions = AccountActions::private(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.is_valid(message_hash)
+    }
+
+    #[aztec(public)]
+    fn is_valid_public(message_hash: Field) -> Field {
+        let actions = AccountActions::public(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.is_valid_public(message_hash)
+    }
+
+    #[aztec(private)]
+    fn set_is_valid_storage(message_hash: Field, value: Field) {
```

Comment:
Following the note from earlier. I think we should be able to just have the internal and then use the entry-point to enter it. Would be pretty clean I think 👀. But let us look into that in a separate pr to not delay this one to run into merge hell.